evaluation_script/main.py

```diff
@@ -53,32 +87,39 @@ def calculate_base_metrics(infile, truth):
             arr = line.split("\t")
             length = len(arr)
             doc_id = arr[0]
-            for i in range(1, length):
-                query_id = index[i]
-                if (query_id, doc_id) in truth:
-                    if (query_id, doc_id) not in predicted_keys:
-                        predicted_keys.add((query_id, doc_id))
-                        if truth[(query_id, doc_id)] == arr[i]:
-                            if arr[i] == '1':
-                                tp = tp + 1
-                            else:
-                                tn = tn + 1
-                        else:
-                            if truth[(query_id, doc_id)] == '1':
-                                fn = fn + 1
+            if doc_id in documents_with_ground_truth:
+                for i in range(1, length):
+                    query_id = index[i]
+                    if (query_id, doc_id) in truth:
+                        if (query_id, doc_id) not in predicted_keys:
```

Comment:
The `if (query_id, doc_id) not in predicted_keys:` test is protecting against duplicated rows (duplicate docids), is that correct?

Reply:
Yes.